encode.go

```diff
@@ -169,7 +189,29 @@ func structEncoder(v reflect.Value) ([]byte, error) {
 		if err != nil {
 			return nil, err
 		}
-		copy(dst[spec.startPos-1:spec.endPos:spec.endPos], val)
+
+		var fieldLen = spec.endPos - spec.startPos + 1
+		if len(val) > fieldLen {
```

Comment:
This is another breaking change in behavior -- could make this opt-in as well. Thoughts? The current behavior is to silently ignore overflow, right?

Reply:
I just added #33 to address it. It should definitely be opt-in.